packages/binding-mqtt/src/mqtt-broker-server.ts

```diff
@@ -102,255 +95,278 @@ export default class MqttBrokerServer implements ProtocolServer {
-                                        );
-                                    }
-
-                                    // topic found and message processed
-                                    return;
-                                } else {
-                                    console.warn(
-                                        "[binding-mqtt]",
-                                        `MqttBrokerServer at ${this.brokerURI} received message for readOnly property at '${receivedTopic}'`
-                                    );
-                                    return;
-                                } // property is writeable? Not necessary since it didn't actually subscribe to this topic
-                            } // Property exists?
-                        }
-                    }
-                    return;
-                }
-                // topic not found
-                console.warn(
-                    "[binding-mqtt]",
-                    `MqttBrokerServer at ${this.brokerURI} received message for invalid topic '${receivedTopic}'`
-                );
-            }
-        );
+        this.broker.publish(name, JSON.stringify(thing.getThingDescription()), { retain: true });
+    }
 
-        for (const eventName in thing.events) {
-            const topic = encodeURIComponent(name) + "/events/" + encodeURIComponent(eventName);
-            const event = thing.events[eventName];
+    private exposeProperty(name: string, propertyName: string, thing: ExposedThing) {
+        const topic = encodeURIComponent(name) + "/properties/" + encodeURIComponent(propertyName);
+        const property = thing.properties[propertyName];
 
+        if (!property.writeOnly) {
             const href = this.brokerURI + "/" + topic;
             const form = new TD.Form(href, ContentSerdes.DEFAULT);
-            form.op = ["subscribeevent", "unsubscribeevent"];
-            event.forms.push(form);
+            form.op = ["readproperty", "observeproperty", "unobserveproperty"];
+            property.forms.push(form);
             console.debug(
                 "[binding-mqtt]",
-                `MqttBrokerServer at ${this.brokerURI} assigns '${href}' to Event '${eventName}'`
+                `MqttBrokerServer at ${this.brokerURI} assigns '${href}' to property '${propertyName}'`
             );
 
-            const eventListener = async (content: Content) => {
-                if (!content) {
+            const observeListener = async (data: Content) => {
+                let content;
+                try {
+                    content = ContentSerdes.get().valueToContent(data, property.data);
+                } catch (err) {
                     console.warn(
                         "[binding-mqtt]",
-                        `HttpServer on port ${this.getPort()} cannot process data for Event ${eventName}`
+                        `MqttServer cannot process data for Property '${propertyName}': ${err.message}`
```

Comment:
@relu91 there is one more HttpServer log that should be fixed in this file.

Reply:
Thank you! good catch! 👍🏻